griptape/common/action.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+import json
+from typing import TYPE_CHECKING, Optional
+
+from attrs import define, field
+
+from griptape.artifacts import BaseArtifact
+from griptape.mixins import SerializableMixin
+
+if TYPE_CHECKING:
+    from griptape.tools import BaseTool
+
+
+@define(kw_only=True)
+class Action(SerializableMixin):
```

Comment:
Should we add some prefix to this? I'm worried that we might have some other "action" in the future, unrelated to LLM actions, which will be confusing naming-wise.

Reply:
`ToolAction`?

Reply:
Refactored into `BaseAction` and `ToolAction`